packages/parser/src/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { JSXElement, createMemo } from "solid-js";
+
+export function createParser(id?: string) {
+  const $TOKEN = Symbol(id || "solid-parser");
+
+  function tokenize<
+    TToken extends Meta & {
+      props: unknown;
+    }
+  >(meta: Omit<TToken, "props">): (props: TToken["props"]) => JSXElement {
+    return (props: any) => {
+      return Object.assign(
+        () => {
+          console.info("tokens can only be rendered inside a Parser");
+          return <></>;
+        },
+        {
+          [$TOKEN]: true,
+          ...meta,
+          props
+        }
+      );
+    };
+  }
+
+  function childrenTokens<T>(children: () => JSXElement | JSXElement[] | T | T[]) {
+    return createMemo(() =>
+      children
+        ? ([] as any[])
+            .concat(children())
+            .filter(child => child && $TOKEN in child)
+            .map((a: T) => a as T)
+        : []
+    );
+  }
```

Comment:
This probably should use the `children` helper from solid to resolve children recursively from functions/arrays
```suggestion
function childrenTokens<T>(fn: () => JSXElement | JSXElement[] | T | T[]) {
  const resolved = children(fn)
  return createMemo(() => resolved.asArray().filter(child => child && $TOKEN in child))
}
```

Reply:
On second thought the `children` helper would also resolve tokenized functions...
I guess this needs to resolve children manually to call normal functions and arrays, but leave the tokens.

Reply:
this is what I initially tried as well, but then it would try to render it out and we get `tokens can only be rendered inside a Parser`

Reply:
this is what I initially tried as well, but then it would try to render it out and we get `tokens can only be rendered inside a Parser`

Reply:
a weird, ur comment didn't show up before. yes exactly!